discovery-provider/src/queries/queries.py

```diff
@@ -188,8 +186,17 @@ def get_tracks_including_unlisted():
                 )
 
         # Perform the query
+        # TODO: pagination is broken with unlisted tracks
         query_results = paginate_query(base_query).all()
         tracks = helpers.query_result_to_list(query_results)
+
+        # Mapping of track_id -> track; used to check route_id when iterating through identifiers
+        track_map = {t["track_id"]: t for t in tracks}
+
+        # If the track is unlisted and the generated route_id does not match the route_id in db,
+        # filter track out from response
+        tracks = [track_map[i["id"]] for i in identifiers if not (track_map[i["id"]]["is_unlisted"] and \
```

Comment:
Logic looks good! I liked in the previous version that you pulled out a function as a predicate for this list comprehension. It gets pretty clunky to constantly refer to the track as `track_map[i["id"]]` etc. It would be nice to have a function that simply took in `i`, pulled out the track from the tracks map, and then tested it with your boolean expression that you have here.